R/zchunk_LA124.heat.R

```diff
@@ -69,117 +77,121 @@ module_energy_LA124.heat <- function(command, ...) {
+      # Select only technologies that have heat output in calibrated techs mapping
       filter(technology %in% calibrated_techs$technology[calibrated_techs$secondary.output == "heat"]) %>%
-      gather(key=year, value=value, -fuel, -sector, -GCAM_region_ID, -technology) %>%
-      mutate(year = sub("X", "", year)) %>%
-      mutate(year = as.integer(year)) %>%
       left_join(L124.out_EJ_R_heatfromelec_F_Yh %>%
-                  rename(value_elec=value) %>%
-                  rename(temp=sector), by=c("GCAM_region_ID", "fuel", "year")) %>%
-      mutate(value=value_elec/value) %>%
-      select(-value_elec, -temp)-> L124.heatoutratio_R_elec_F_tech_Yh
+                  rename(value_elec = value) %>%
+                  rename(temp = sector), by=c("GCAM_region_ID", "fuel", "year")) %>%
+      mutate(value = value_elec / value) %>%
+      select(GCAM_region_ID, sector, fuel, technology, year, value) %>%
+      # Reset missing and infinite values (applicable for CC in the base years) to 0
+      mutate(value = if_else(is.na(value)|is.infinite(value), 0, value)) -> L124.heatoutratio_R_elec_F_tech_Yh
 
-    #Reset missing and infinite values (applicable for CC in the base years) to 0
-    L124.heatoutratio_R_elec_F_tech_Yh %>%
-      mutate(value=if_else(is.na(value)|is.infinite(value), 0, value))-> L124.heatoutratio_R_elec_F_tech_Yh
 
-    #Drop all rows where all years are 0
+    # Drop all rows where all years are 0
+    # Create a table of all the years of all 0
     L124.heatoutratio_R_elec_F_tech_Yh %>%
       group_by(GCAM_region_ID, technology, sector, fuel) %>%
-      summarise(sum=sum(value)) %>%
-      filter(sum==0) -> all_years_0
+      summarise(sum = sum(value)) %>%
+      filter(sum == 0) -> all_years_0
 
+    # Filter out the years in all_years_0 from heatoutratio
     L124.heatoutratio_R_elec_F_tech_Yh %>%
       left_join(all_years_0) %>%
-      mutate(sum=if_else(is.na(sum), 2, sum)) %>% #2 was a random, nonzero choice
-      filter(sum!=0)  %>%
+      #2 is a random, non-zero option for temporary use
+      mutate(sum = if_else(is.na(sum), 2, sum)) %>%
+      filter(sum != 0)  %>%
       select(-sum) -> L124.heatoutratio_R_elec_F_tech_Yh
 
     # This is complicated. Some historical regions / years have 0 district heat output (all fuels). This is fine unless there is heat output from
     # IEA's MAINCHP flow, which we model in the electric sector with a secondary output of heat to the district heat sector. In GCAM this will
     # cause solution failure as the district heat supply will be 0. In order to avoid this outcome we need a nominal amount of heat produced
     # in the zero years. The code below sets an exogenous year as the one to use to extract fuel shares for each region. In all 0 region / years,
     # output is equal to the output in the fuel-share year times 1e-3 to avoid throwing off the energy balances.
-    fuel_share_year <- "2010"
+
+    fuel_share_year <- 2010
```

Comment:
fuel_share_year and later fsy gets very confusing. I would prefer historical_year and hist_value or norm_year or norm_value (normalized).